storage/azure/src/main/java/io/aiven/kafka/tieredstorage/storage/azure/AzureBlobStorage.java

```diff
@@ -69,6 +74,16 @@ public void configure(final Map<String, ?> configs) {
 
         metricsPolicy = new MetricCollector(config);
 
+        final ProxyConfig proxyConfig = config.proxyConfig();
+        if (proxyConfig != null) {
+            proxyOptions = new ProxyOptions(ProxyOptions.Type.SOCKS5,
+                new InetSocketAddress(proxyConfig.host(), proxyConfig.port()));
+            if (proxyConfig.username() != null) {
+                proxyOptions.setCredentials(proxyConfig.username(), proxyConfig.password());
+            }
+            blobServiceClientBuilder.clientOptions(new HttpClientOptions().setProxyOptions(proxyOptions));
+        }
```

Comment:
Maybe move the proxy options initialization into `AzureBlobStorageConfig`? And then just have something like this here?
`config.proxyOptions()
            .ifPresent(options -> blobServiceClientBuilder.clientOptions(new HttpClientOptions().setProxyOptions(
                options)));`

Reply:
I thought we were moving the opposite direction: removing the logic from `...Config` classes, no? :)